### PR TITLE
TurtleShroom Productions III !

### PR DESCRIPTION
**Pull request by TurtleShroom**

### CODE REVIEW

**Comment by TurtleShroom** [comment]
@loup99 

Here is the latest batch of works I've produced. Enjoy!

File: WTWSMS/common/landed_titles/mercenaries.txt [hunk 1]
```diff
@@ -715,40 +715,6 @@ d_sunni_cuman_company = {
...
-	primary = yes
-	
-	# Cannot be vassalized
-	independent = yes
-	
-	strength_growth_per_century = 1.00
-	
-	mercenary_type = berber_company_composition
-}
-
```
Comment (loup99) [inline WTWSMS/common/landed_titles/mercenaries.txt]: Why did you move these mercenaries?

File: WTWSMS/decisions/VIET_Eastern_religions_decisions.txt [hunk 2]
```diff
@@ -31,7 +32,7 @@ decisions = {
 					trait = kind
 				}
 			}
-			NOR = {
+			NOT = {
```
Comment (loup99) [inline WTWSMS/decisions/VIET_Eastern_religions_decisions.txt]: Why did you change this?

File: WTWSMS/decisions/WtWSMS_realm_decisions.txt [hunk 3]
```diff
@@ -350,6 +350,10 @@ decisions = {
 		effect = {
 			set_global_flag = rome_restored
 			narrative_event = { id = ancrel.2220 tooltip = EVTTOOLTIPLoR.20 }
+			add_law = succ_primogeniture
+			add_law = vice_royalty_2
+			add_law = imperial_administration
+			set_allow_free_vice_royalty_revokation = yes
```
Comment (loup99) [inline WTWSMS/decisions/WtWSMS_realm_decisions.txt]: This gives it freely though.

File: WTWSMS/localisation/WtWSMS_religions.csv [hunk 4]
```diff
@@ -737,6 +739,15 @@ HYMNBOOK;The Black Hymnal;;;;;;;;;;x
...
+k_mimamsa_ADJ;Mimamsa;;;;;;;;;;;;;x
+k_melchisedechianism;Salem;;;;;;;;;;;;;x
+k_melchisedechianism_ADJ;Melchisedekite;;;;;;;;;;;;;x
+SALEM;King-Priest;;;;;;;;;;;;;x
+d_melchisedechianism_holy_order;Merkabah Riders;;;;;;;;;;;;;x
+d_melchisedechianism_holy_order_ADJ;Merkabanite;;;;;;;;;;;;;x
+create_salem;Instate Salem;;;;;;;;;;;;;x
+create_salem_desc;Melchisedechianism has always been dismissed as a strange, mysticist splinter of Christianity. No more! We are now in the position to restore the ancient Kingdom of Salem, that Father Abraham himself gave his wealth to, and glorify God's name as He deserves once more! Glory to Salem, glory to El-ʿElyōn!;;;;;;;;;;;;;x
+create_melchisedechianism_holy_order;Form the Order of Melchizedek;;;;;;;;;;;;;x
+create_melchisedechianism_holy_order_desc;With the Kingdom of Salem in place, we can now begin to focus on defending the faith from the onset of heathenry and paganry that we face every day. Let us convene the most zealous, devoted, and brilliant strategists to defend the followers of Melchezedek wherever they live!;;;;;;;;;;;;;x
```
Comment (loup99) [inline WTWSMS/localisation/WtWSMS_religions.csv]: No decisions should be in this file.

File: WTWSMS/localisation/WtWSMS_religions_unused.csv [hunk 5]
```diff
@@ -56,8 +56,6 @@ GOD_OMNIPATER;The Omnipater;;;;;;;;;;;x
 GOD_LUXIMATER;The Luximater;;;;;;;;;;;x
 GOD_THE_FROGMEN;The Frogmen of Wasketville Court;;;;;;;;;;;x
 GOD_CONNOR_ANDREW_DAY;Connor Andrew Day;;;;;;;;;;;x
-banditism;Banditism;;;;;;;;;;;x
-banditism_DESC;The highly heretical Bandit sect claims that the almighty Delorius is not a unicorn, but instead a regular horse named Bandit. Definitely an Athiest sect, thanks Communist Obama!;;;;;;;;;;;x
```
Comment (loup99) [inline WTWSMS/localisation/WtWSMS_religions_unused.csv]: I get most of it, and this file is unused anyway.

File: WTWSMS/common/landed_titles/mercenaries.txt (hunk 1 above)
Comment (TurtleShroom) [inline WTWSMS/common/landed_titles/mercenaries.txt]: I moved the "Knights of the VALENTINIANS" over to the Holy Orders because it's a Valentinian Holy Order.

File: WTWSMS/decisions/VIET_Eastern_religions_decisions.txt (hunk 2 above)
Comment (TurtleShroom) [inline WTWSMS/decisions/VIET_Eastern_religions_decisions.txt]: @loup99 

It was an error. "NOT" tells the code that if any one of those conditions are met, the action won't be applied or enabled. "NOR" is not a command. It was a typo.

File: WTWSMS/decisions/WtWSMS_realm_decisions.txt (hunk 3 above)
Comment (TurtleShroom) [inline WTWSMS/decisions/WtWSMS_realm_decisions.txt]: I don't understand what you mean.

File: WTWSMS/localisation/WtWSMS_religions.csv (hunk 4 above)
Comment (TurtleShroom) [inline WTWSMS/localisation/WtWSMS_religions.csv]: @loup99 

Gladly. I will fix it immediately.

File: WTWSMS/decisions/VIET_Eastern_religions_decisions.txt (hunk 2 above)
Comment (loup99) [inline WTWSMS/decisions/VIET_Eastern_religions_decisions.txt]: NOR is a command.

File: WTWSMS/decisions/WtWSMS_realm_decisions.txt (hunk 3 above)
Comment (loup99) [inline WTWSMS/decisions/WtWSMS_realm_decisions.txt]: I don't want those who restore the roman empire to receive a free boost in terns of laws.

File: WTWSMS/common/cb_types/WtWSMS_migration_unification_cb_types.txt [hunk 6]
```diff
@@ -39,12 +39,23 @@ migration = {
...
+					NOT = {
+						AND = {
+							higher_title_than = KING
+							is_nomadic = no
+						}
+					}
+	
 					NOT = { culture = FROM }
 					independent = yes # No vassals
 					war = no # Ensures that the events activate correctly
```
Comment (loup99) [inline WTWSMS/common/cb_types/WtWSMS_migration_unification_cb_types.txt]: I disagree on this change. Migration is limited with a purpose, and you both expand it religion-wise and disable it for all non-nomads.

File: WTWSMS/common/scripted_effects/WtWSMS_scripted_effects.txt [hunk 7]
```diff
@@ -4,12 +4,13 @@
...
+		religion = LIEGE
 		dynasty = NONE
 		female = no
 		age = 34
 		attributes = {
-			martial = 8
+			martial = 10
 		}
 		trait = skilled_tactician
 	}
```
Comment (loup99) [inline WTWSMS/common/scripted_effects/WtWSMS_scripted_effects.txt]: This will be done but later.

File: WTWSMS/decisions/minor_decisions.txt [hunk 8]
```diff
@@ -3,11 +3,6 @@ decisions = {
...
 		potential = {
 			is_ruler = yes
-			OR = {
-				is_feudal = yes
-				is_tribal = yes
-				is_nomadic = yes
-			}
 			prisoner = no
 			OR = {
 				religion_group = christian
```
Comment (loup99) [inline WTWSMS/decisions/minor_decisions.txt]: This requirement should be kept.

File: WTWSMS/localisation/WtWSMS_religions_unused.csv [hunk 9]
```diff
@@ -241,5 +241,5 @@ dacian_pagan;Zalmoxian;;;;;;;;;;;x
...
 umbrian_pagan;Umbri'i;;;;;;;;;;;x
 umbrian_pagan_DESC;During the 6th�4th centuries BC, Umbrian communities constructed rural sanctuaries in which they sacrificed to the gods. Bronze votives shaped as animals or deities were also offered. Umbrian deities include Feronia, Valentia, Minerva Matusia and Clitumnus. The Iguvine Tablets were discovered in 1444 at Gubbio, Italy. Composed during the 2nd or 3rd centuries BC, they describe religious rituals involving animal sacrifice;;;;;;;;;;;x
-roman_pagan;Romanic;;;;;;;;;;;x
-roman_pagan_DESC;The Romanic heresy is a deviant branch of the Hellenic church the believes the addition of Greek deities corrupted the original Roman faith. Also anti-clerical in nature, the Romanic faith has no religious head or special game mechanics.;;;;;;;;;;;x
\ No newline at end of file
+romanic_pagan;Romanic;;;;;;;;;;;x
+romanic_pagan_DESC;The Romanic heresy is a deviant branch of the Hellenic church the believes the addition of Greek deities corrupted the original Roman faith. Also anti-clerical in nature, the Romanic faith has no religious head or special game mechanics.;;;;;;;;;;;x
```
Comment (loup99) [inline WTWSMS/localisation/WtWSMS_religions_unused.csv]: Leave this file.

File: WTWSMS/common/minor_titles/WtWSMS_imperial_legacy_minor_titles.txt [hunk 10]
```diff
@@ -466,11 +492,12 @@ title_il_comes_sacrae_vestis = {
...
+		OR = {
 			has_landed_title = e_wre
-			culture = roman
+			has_landed_title = e_roman_empire
 		}
-	}		
+	}
 
 	allowed_to_hold = {
 		is_adult = yes
```
Comment (loup99) [inline WTWSMS/common/minor_titles/WtWSMS_imperial_legacy_minor_titles.txt]: The requirement should in my opinion be for Romans.

File: WTWSMS/localisation/ACR_hellenic_events.csv [hunk 11]
```diff
@@ -103,3 +103,10 @@ ancrel.2568.title;The Sacred Band;La Bande Sacr
...
 ancrel.2568.a;Very good!;;;;;;;;;;;;;x
 ancrel.2568.b;This is grave news...;;;;;;;;;;;;;x
+mastimani_human_sacrifice;Feed to Mastiman;;;;;;;;;;;;;x
+mastimani_human_sacrifice;Mastiman is hungry. Mastiman craves the blood of infidels, heretics, and traitors. We shall give him what he desires.;;;;;;;;;;;;;x
+HAIL_SATURN1;Innocent [From.GetManWoman] Sacrificed to Saturn!;;;;;;;;;;;;;x
+HAIL_SATURN2;Mastiman Must Feed!;;;;;;;;;;;;;x
+HAIL_SATURN1_DESCRIPTION;[Root.GetFOA], we have learned that the filthy Mastimanis have sacrificed [From.GetBestName] to their bloodthirsty demons! After setting up an altar and presenting food and drink offerings, [From.GetBestName] was slaughtered and [From.GetHisHer] body lay mutilated before the throngs of cheering faithful.;;;;;;;;;;;;;x
+HAIL_SATURN2_DESCRIPTION;Mastiman is hungry. Mastiman craves blood. Let the Augurs do what must be done. [From.GerBestName] should be honored to fulfill such a noble and holy goal.;;;;;;;;;;;;;x
+FEEDMASTIMAN;BLOOD FOR THE BLOOD GOD: SOULS FOR MASTIMAN! ALL HAIL!;;;;;;;;;;;;;x
```
Comment (loup99) [inline WTWSMS/localisation/ACR_hellenic_events.csv]: I have already explained why this is not necessary.

File: WTWSMS/decisions/VIET_Eastern_religions_decisions.txt (hunk 2 above)
Comment (TurtleShroom) [inline WTWSMS/decisions/VIET_Eastern_religions_decisions.txt]: @loup99 

Isn't that what "NOT" does? If anything in a "NOT" zone is true, the entire thing is rejected. Is "NOR" the opposite?

File: WTWSMS/decisions/WtWSMS_realm_decisions.txt (hunk 3 above)
Comment (TurtleShroom) [inline WTWSMS/decisions/WtWSMS_realm_decisions.txt]: @loup99  

Okay. That still doesn't address why the two Roman Empires don't have the Imperial Administration Law in effect at the game's start, like they would have had in RL. What I wrote about _uniting_ (not restoring) the two Roman Empires was a makeshift quick fix because I had no idea how to code the two empires to start with Imperial Administration.

I'll revert it if you'd like.

File: WTWSMS/localisation/WtWSMS_religions_unused.csv (hunk 5 above)
Comment (TurtleShroom) [inline WTWSMS/localisation/WtWSMS_religions_unused.csv]: Can we still remove it anyway? It irritates me that it's there.

File: WTWSMS/decisions/VIET_Eastern_religions_decisions.txt (hunk 2 above)
Comment (loup99) [inline WTWSMS/decisions/VIET_Eastern_religions_decisions.txt]: NOT returns true if the enclosed condition is false.

File: WTWSMS/decisions/WtWSMS_realm_decisions.txt (hunk 3 above)
Comment (loup99) [inline WTWSMS/decisions/WtWSMS_realm_decisions.txt]: Preferably yes.

File: WTWSMS/common/cb_types/WtWSMS_migration_unification_cb_types.txt (hunk 6 above)
Comment (TurtleShroom) [inline WTWSMS/common/cb_types/WtWSMS_migration_unification_cb_types.txt]: @loup99 

SHOOT! Loup, I'm so sorry. I didn't mean to disable it for everyone except nomads.

What I was trying to code was this:

1. Nomads can migrate, but no other type of Emperor can.
2. Reformed incarnations of the previously available nomads can still migrate. Why should a Slavic or Germanic/Norse Pagan stop doing what they do best?

For the sake of historical accuracy, I also think that Magyars, Iazygens, and Iranian Steppe Pagans should be allowed to migrate freely (as they did in RL), but I decided to save that for another time.

File: WTWSMS/common/scripted_effects/WtWSMS_scripted_effects.txt (hunk 7 above)
Comment (TurtleShroom) [inline WTWSMS/common/scripted_effects/WtWSMS_scripted_effects.txt]: Okay. I'll remove it now.

File: WTWSMS/localisation/WtWSMS_religions_unused.csv (hunk 5 above)
Comment (loup99) [inline WTWSMS/localisation/WtWSMS_religions_unused.csv]: If that irritates you out of everything, WtWSMS is not your can of works. :stuck_out_tongue:

File: WTWSMS/decisions/minor_decisions.txt (hunk 8 above)
Comment (TurtleShroom) [inline WTWSMS/decisions/minor_decisions.txt]: Why can't any government meeting the religious requirements hold a feast? Are you telling me that wealthy Merchant Republics didn't pig out from time to time?

Comment (loup99) [inline WTWSMS/decisions/minor_decisions.txt]: Not in the way this one works.

File: WTWSMS/common/minor_titles/WtWSMS_imperial_legacy_minor_titles.txt (hunk 10 above)
Comment (TurtleShroom) [inline WTWSMS/common/minor_titles/WtWSMS_imperial_legacy_minor_titles.txt]: I didn't change that. What I added was that the Restored Roman Empire could use ERE AND WRE titles. Everything else stayed the same (must be Roman culture, must hold RE, WRE, or ERE, etc.).

**Comment by TurtleShroom** [comment]
@loup99 

Okay, I reverted the Mastimani addition because it has already been added. I also reverted the "NOT" back to "NOR".

Is that what you wanted me to do?

File: WTWSMS/localisation/ACR_hellenic_events.csv (hunk 11 above)
Comment (TurtleShroom) [inline WTWSMS/localisation/ACR_hellenic_events.csv]: Darn. I was too slow.

File: WTWSMS/localisation/WtWSMS_religions_unused.csv (hunk 5 above)
Comment (TurtleShroom) [inline WTWSMS/localisation/WtWSMS_religions_unused.csv]: It irritates me so badly because it's an incorrect use of the "THANKS OBAMA" meme.

File: WTWSMS/common/minor_titles/WtWSMS_imperial_legacy_minor_titles.txt (hunk 10 above)
Comment (loup99) [inline WTWSMS/common/minor_titles/WtWSMS_imperial_legacy_minor_titles.txt]: Why did you remove culture = roman?

File: WTWSMS/common/cb_types/WtWSMS_migration_unification_cb_types.txt (hunk 6 above)
Comment (loup99) [inline WTWSMS/common/cb_types/WtWSMS_migration_unification_cb_types.txt]: I think the current setup should be enough for 1 to not be an issue. On 2, that is a design question. I disagree with you and think reformed pagans are more settled than the unreformed versions.

**Comment by loup99** [comment]
Could you maybe redo this one? There are conflicts with master and most of it is either obsolete/undesired. I will also do compatibility, so it will just be less compatible.
